### ISSUE TITLE
getting the predict class ?

### ISSUE THREAD

**Issue by anujgupta82** [LIBOL/SOL#1]
I am using python wrapper to use LIBSOL. Once I train a model (python python/libsol_train.py -m arow.model data/a1a arow2.model), **how do I access/return the class predict by the  trained model for a unseen data point ?** 

I only see test (https://github.com/LIBOL/LIBSOL/blob/master/python/libsol_core.py#L170) which internally calls _LIB.lsol_Test(https://github.com/LIBOL/LIBSOL/blob/master/python/libsol_core.py#L186)

**Comment by yuewu001** [comment]
the third parameter "output_path" is the data path to save the predicted classes and scores. We are also working on new wrappers so that you can get the predicted classes and scores from test data of numpy arrays or scipy sparse arrays.

**Comment by anujgupta82** [comment]
third parameter "output_path" is not working. Please see #3

**Comment by anujgupta82** [comment]
Can I know what kind of timeline can I except for predictions/scoring on numpy arrays or scipy sparse arrays ?

**Comment by yuewu001** [comment]
i'll release a new version in the dev branch at the end of this week, which features the usage like sklearn.

**Comment by anujgupta82** [comment]
Cool. Couple of suggestions (based on PA from sklearn) you may consider:

1) ability to add 'class weights' to handle class imbalance while initializing the model
2) fit() : do batch training on data if need be. Returns model and training loss incurred.  
3) partial_fit(): update the current model while fitting a new data point
4) predict() : predict class/label for given data point(s)
5) score() : accuracy score on the test set
6) confidence() : confidence in prediction (usually distance from decision boundary)
7) access model parameters
8) should not necessary read data and write back results to files. ability to accept and return Numpy arrays will be of great help

**Comment by yuewu001** [comment]
thank you so much for your kind suggestion. You can track the 'cython' branch, since the updates in that branch is mostly similar to what you have mentioned.
As to the class imbalance problem, we are still thinking on a algorithmic way for the problem. Your suggestion is still of great value.

**Comment by yuewu001** [comment]
@anujgupta82 i have majorly finished the new python interfaces in the cython branch. I'm looking forward to your suggestions and bug reports if you have time to have a try. thanks
